### ISSUE TITLE
Changing the sequence length for BERT & performance on data with multi-class labels.

### ISSUE THREAD

**Issue by abhinivesh-s** [GT-SALT/MixText#23]
I've been working on this and got results close to what was mentioned in the paper for the IMDB dataset.
However, i'm facing a couple of issues.

1. Changing the sequence length that BERT accepts hasn't been straightforward. I want it changed to 128. I couldn't see that in 'mixtext.py'. 'normal_bert.py' contains 'length=256' in the definition for 'forward' but I'm unsure if this the one to change as I don't see 'length' being used anywhere else. 
Please let me know how to change the sequence length for both normal_bert and mix_text.

2. I tried to run this on the sklearn's NewsGroup dataset - which contains 20 labels. I've appropriately changed the 'read_data.py' to take in this data & also performed back-translations as mentioned. But I couldn't get good results with 'MixText'. Whatever the amount of labeled data that I use, I get almost similar, very low accuracies. Training time also remains the same irrespective of what the labeled data is.
I'm assuming there is no issue with my data reading & pre-processing because 'normal_bert' runs fine.
Please do let me know if there is any known issue with using MixText for a multi-class dataset. 

Thanks!

**Comment by jiaaoc** [comment]
1. The sequences length should be passed into data pre-processing (tokenization)
2. For 20 labels, you might also want to tune the hyper-parameters like the sharpening temperature.

**Comment by abhinivesh-s** [comment]
Hi.
Thanks a lot for the reply!

Just to confirm, I see max_seq_len in read_data.py and I'll change that value. But I also see '256' to be hard-coded at some places in train.py.
For example, here:

                    'if length1 + length2 > 256:
                        length2 = 256-length1 - 1'

https://github.com/GT-SALT/MixText/blob/f17198d98e1bbb012d8e33cd26e228dd43bb4673/code/train.py#L308-L309

Please do let me know if I have to make any such additional changes.
Thanks!

**Comment by jiaaoc** [comment]
Yes, this part in the train.py is actually not used. But you could definitely make modifications to them.
